src/components/Message/MessageDeleted.js

```diff
@@ -1,38 +1,38 @@
 // @ts-check
-import React, { PureComponent } from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
-import { withTranslationContext } from '../../context';
+import { TranslationContext } from '../../context';
+import { useUserRole } from './hooks/useUserRole';
 
 /**
- * @typedef { import('types').MessageDeletedProps } Props
- * @extends PureComponent<Props>
+ * @type{React.FC<import('types').MessageDeletedProps>}
  */
-class MessageDeleted extends PureComponent {
-  static propTypes = {
-    /** The [message object](https://getstream.io/chat/docs/#message_format) */
-    message: PropTypes.object,
-    /** Returns true if message belongs to current user */
-    isMyMessage: PropTypes.func,
-  };
+const MessageDeleted = ({ message }) => {
+  const { t } = useContext(TranslationContext);
+  const { isMyMessage } = useUserRole(message);
+  const messageClasses = isMyMessage
+    ? 'str-chat__message str-chat__message--me str-chat__message-simple str-chat__message-simple--me'
+    : 'str-chat__message str-chat__message-simple';
 
-  render() {
-    const { isMyMessage, message, t } = this.props;
-    const messageClasses = isMyMessage(message)
-      ? 'str-chat__message str-chat__message--me str-chat__message-simple str-chat__message-simple--me'
-      : 'str-chat__message str-chat__message-simple';
-
-    return (
-      <div
-        key={message.id}
-        className={`${messageClasses} str-chat__message--deleted ${message.type} `}
-        data-testid={'message-deleted-component'}
-      >
-        <div className="str-chat__message--deleted-inner">
-          {t && t('This message was deleted...')}
-        </div>
+  return (
+    <div
+      key={message.id}
+      className={`${messageClasses} str-chat__message--deleted ${message.type} `}
+      data-testid={'message-deleted-component'}
+    >
+      <div className="str-chat__message--deleted-inner">
+        {t && t('This message was deleted...')}
       </div>
-    );
-  }
-}
+    </div>
+  );
+};
 
-export default withTranslationContext(MessageDeleted);
+MessageDeleted.propTypes = {
+  /** The [message object](https://getstream.io/chat/docs/#message_format) */
+  // @ts-ignore
```

Comment:
I'll try to make this one work without ignoring it

Reply:
what's the error you're getting from this one?

Reply:
Typescript was matching its definition of the message object against the more loose `propType` one and complaining about it. Using a [more strict](https://github.com/GetStream/stream-chat-react/pull/338/files#diff-91e6dd3754c70f9237d1963e304a426bR110) definition fixed it. I amended the commit with the fix.